pixleesdk/src/main/java/com/pixlee/pixleesdk/PXLAlbumFilterOptions.java

```diff
@@ -53,14 +53,17 @@ public String toParamString() {
             jsonFilters.putOpt("filter_by_radius", filterByRadius);
             //TODO: handle the arrays and dates properly
             if (contentSource != null && contentSource.size() > 0) {
+                boolean isInstagramAdded = false;
                 JSONArray sources = new JSONArray();
                 for (int i = 0; i < contentSource.size(); i++) {
                     sources.put(contentSource.get(i).value);
 
                     // if instagram_feed or instagram_story is added here, instagram has to be added together.
                     // this is a rule. For your understanding, please read the comment of 'content_source' on this document: https://developers.pixlee.com/reference#consuming-content
-                    if (PXLContentSource.INSTAGRAM_FEED == contentSource.get(i) ||
-                            PXLContentSource.INSTAGRAM_STORY == contentSource.get(i)) {
+                    if (!isInstagramAdded &&
+                            (PXLContentSource.INSTAGRAM_FEED == contentSource.get(i) ||
```

Comment:
you could probably just do this check after the loop to see if `PXLContentSource.INSTAGRAM_FEED` or `PXLContentSource.INSTAGRAM_STORY` are in the `contentSource` array , that way we don't need the `isInstagramAdded` flag

Reply:
That can be an option to do this work. To check the `sources` array after the for loop, another loop is needed to check the exitance of 'instagram'. I just want to avoid using another loop. Then, we can add parameters and check them within a single loop. But since the size of the array is small(less than 10), using one array does not have benefits for the runtime.  I think either way is totally fine. How do you think? 

## Current
```
boolean isInstagramAdded = false;
JSONArray sources = new JSONArray();
for (int i = 0; i < contentSource.size(); i++) {
    sources.put(contentSource.get(i).value);
    if (!isInstagramAdded &&
            (PXLContentSource.INSTAGRAM_FEED == contentSource.get(i) ||
                    PXLContentSource.INSTAGRAM_STORY == contentSource.get(i))) {
        isInstagramAdded = true;
        sources.put("instagram");
    }
}
```

## KB's suggestion
```
JSONArray sources = new JSONArray();
for (int i = 0; i < contentSource.size(); i++) {
    sources.put(contentSource.get(i).value);
}

for (int i = 0; i < sources.length(); i++) {
    if (PXLContentSource.INSTAGRAM_FEED == contentSource.get(i) ||
            PXLContentSource.INSTAGRAM_STORY == contentSource.get(i)) {
        sources.put("instagram");
        break;
    }
}
```

Btw, this code is to avoid this data.
Problem: ["instagram_feed","instagram","instagram_story","instagram"]
Solution: ["instagram_feed","instagram","instagram_story"]